File: courses/C# 2008 - Impacta/C# 2008 - M2/Instrutor/CSModII20090918/VB.NET2008 Mod2 Hanna/Hanna VBNet II 03082009/Projetos/prjCEP/prjCEP/Geral.vb
Public Class Geral
    Public Shared Function TestarVazio(ByVal txt As TextBox) _
    As String
        If txt.Text.Trim() = "" Then
            txt.Focus()
            txt.SelectAll()
            Throw New Exception("Preencha o campo - " & _
            txt.Name.Replace("TextBox", "").ToUpper())
        End If
        Return txt.Text
    End Function
End Class

File: courses/C# 2008 - Impacta/C# 2008 - M1/16.MDI/prjOficina/prjOficina/Geral.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using System.Windows.Forms;

namespace prjOficina
{
    class Geral
    {
        public static short EhShort(TextBox txt)
        {
            try
            {
                return Convert.ToInt16(txt.Text);
            }
            catch (Exception)
            {
                txt.Focus();
                txt.SelectAll();
                throw new Exception("Informe um ano válido");
            }
        }

        public static string TahVazio(TextBox txt)
        {
            if (txt.Text.Trim() != "")
            {
                return txt.Text;
            }
            else
            {
                txt.Focus();
                txt.SelectAll();
                throw new Exception("Preencha " 
                    + txt.Name.Substring(0, txt.Name.Length-7));
            }
        }

        public static void LimparTela(Form formulario)
        {
            foreach (Control ctl in formulario.Controls)
            {
                if (ctl is TextBox)
                {
                    //ctl.Text = "";
                    (ctl as TextBox).Clear();
                }
            }
        }

    }
}
